src/class/dfu/dfu_device.c

```diff
@@ -249,6 +249,7 @@ static uint16_t dfu_req_upload(uint8_t rhport, tusb_control_request_t const * re
 {
   TU_VERIFY( wLength <= CFG_TUD_DFU_TRANSFER_BUFFER_SIZE);
   uint16_t retval = tud_dfu_req_upload_data_cb(block_num, (uint8_t *)_dfu_state_ctx.transfer_buf, wLength);
+  TU_ASSERT( retval <= CFG_TUD_DFU_TRANSFER_BUFFER_SIZE);
```

Comment:
I think this is not really needed, since the transfer is limited by wLength which we already TU_VERIFY as above. The usbd_control will truncate any data exceeding the wLength